rplugin/python3/deoplete/sources/deoplete_jedi/server.py

```diff
@@ -476,15 +476,7 @@ def __init__(self, desc_len=0, short_types=False, show_docstring=False,
                  os.path.dirname(os.path.dirname(__file__)))),
         })
 
-        if 'VIRTUAL_ENV' in os.environ:
-            self.env['VIRTUAL_ENV'] = os.getenv('VIRTUAL_ENV')
-            prog = os.path.join(self.env['VIRTUAL_ENV'], 'bin', 'python')
-        elif python_path:
-            prog = python_path
-        else:
-            prog = 'python'
-
-        self.cmd = [prog, '-u', __file__, '--desc-length', str(desc_len)]
```

Comment:
Not sure if moving the `$VIRTUAL_ENV` check breaks any use case, e.g. changing `$VIRTUAL_ENV` and then restarting the Server?

It will not pass on `$VIRTUAL_ENV` via `env=self.env` in `restart` anymore, but that seems to not be required anyway, is it?

Reply:
I don't know why the virtualenv check exists.